src/QsCompiler/Transformations/CallGraph.cs

```diff
@@ -192,30 +170,344 @@ public override int GetHashCode()
+
+                        if (accum.TryGetValue(dependent, out var existingEdges))
+                        {
+                            combinedEdges = combinedEdges.Where(edge => !existingEdges.Any(existing => edge.Equals(existing)));
+                            accum[dependent] = existingEdges.AddRange(combinedEdges);
+                        }
+                        else
+                        {
+                            accum[dependent] = combinedEdges.ToImmutableArray();
+                        }
+
+                        foreach (var edge in combinedEdges)
+                        {
+                            WalkDependencyTree(dependent, edge);
+                        }
+                    }
+                }
+            }
+
+            WalkDependencyTree(callerSpec, null);
+
+            return accum
+                .SelectMany(kvp => kvp.Value, Tuple.Create)
+                .ToLookup(tup => tup.Item1.Key, tup => tup.Item2);
+        }
+
+        /// <summary>
+        /// Returns all specializations that are used directly or indirectly within the given caller,
+        /// whether they are called, partially applied, or assigned. Each key in the returned lookup
+        /// represents a specialization that is used by the caller. Each value in the lookup is an
+        /// IEnumerable of edges representing all the different ways the given caller specialization took a
+        /// dependency on the specialization represented by the associated key.
+        /// Throws ArgumentNullException if argument is null.
+        /// Returns an empty ILookup if the node was found with no dependencies or was not found in
+        /// the graph.
+        /// </summary>
+        public ILookup<ICallGraphNode, ICallGraphEdge> GetAllDependencies(QsSpecialization callerSpec)
+        {
+            if (callerSpec == null)
+            {
+                throw new ArgumentNullException(nameof(callerSpec));
+            }
+
+            return this.GetAllDependencies(new CallGraphNode(callerSpec.Parent, callerSpec.Kind, callerSpec.TypeArguments));
+        }
+
+        /// <summary>
+        /// Returns all specializations that are used directly within the given caller, whether they are
+        /// called, partially applied, or assigned. Each key in the returned lookup represents a
+        /// specialization that is used by the caller. Each value in the lookup is an IEnumerable of edges
+        /// representing all the different ways the given caller specialization took a dependency on the
+        /// specialization represented by the associated key.
+        /// Throws ArgumentNullException if argument is null.
+        /// Returns an empty ILookup if the node was found with no dependencies or was not found in
+        /// the graph.
+        /// </summary>
+        public ILookup<ICallGraphNode, ICallGraphEdge> GetDirectDependencies(ICallGraphNode callerNode)
```

Comment:
I think you could replace these doc comments with `/// <inheritdoc/>` since the interface has documentation now.

Reply:
The interfaces have been removed, so this is the only place where the doc comments live now.